odoo/models.py

```diff
@@ -3002,6 +3002,10 @@ def _update_field_translations(self, field_name, translations, digest=None):
         """
         self.ensure_one()
 
+        installed_lang = list(code for code, _ in self.env['res.lang'].get_installed())
+        if set(translations) - set(installed_lang):
+            raise UserError(_("The following language is not activated: %s", (set(translations) - set(installed_lang))))
```

Comment:
The following languages are not installed: %s

Another thing to be discussed:
Could you tell me when the user will trigger this UserError from the UI?
I think the only use case is: user A is translating, user B deactivates one language before user A saves the changes. In that case I think we can just drop invalid languages and keep it silent. So the user don't have to refresh the page and translate again.

And you don't have to export the new sentence to the pot